data/users.js

```diff
@@ -79,93 +82,141 @@ let exportedMethods = {
   },
-  
-  async addOwnedLibrary(userId,libraryId){
 
-    libraryId = validation.checkValidId(libraryId, "Library ID")
-    userId = validation.checkValidId(userId, "User ID")
+  async addOwnedLibrary(userId, libraryId) {
+    libraryId = validation.checkValidId(libraryId, "Library ID");
+    userId = validation.checkValidId(userId, "User ID");
     const userCollection = await users();
 
     const user = await userCollection.findOne({ _id: new ObjectId(userId) });
     if (user === null) throw "VError: No user found with given ID.";
-    
-    // If the user has already favorited this 
-    if (user.ownedLibraries.includes(libraryId)) throw "VError: User has already owns this library"
-    
+
+    // If the user has already favorited this
+    if (user.ownedLibraries.includes(libraryId))
+      throw "VError: User has already owns this library";
+
     await userCollection.updateOne(
       { _id: new ObjectId(userId) },
-      { $push: {ownedLibraries: libraryId} }
-      );
+      { $push: { ownedLibraries: libraryId } }
+    );
   },
 
+  /**
+   * @name dropOwnedLibrary
+   * @author jcarr2
+   * @param userId
+   * @param libraryId
+   */
+  async dropOwnedLibrary(userId, libraryId) {
+    libraryId = validation.checkValidId(libraryId, "Library ID");
+    userId = validation.checkValidId(userId, "User ID");
+    const userCollection = await users();
+
+    const user = await userCollection.findOne({ _id: new ObjectId(userId) });
+    if (user === null) throw "VError: No user found with given ID.";
+
+    if (!user.ownedLibraries.includes(libraryId))
+      throw "VError: User does not own this library";
+
+    try {
+      await userCollection.updateOne(
+        { _id: new ObjectId(userId) },
+        { $pull: { ownedLibraries: libraryId } }
+      );
+    } catch (e) {
+      throw "VError: Could not update user";
+    }
+  },
   /**
    * This function will validate the given user credentials
-   * @param {string} emailAddress 
-   * @param {string} password 
+   * @param {string} emailAddress
+   * @param {string} password
    */
   async checkUser(emailAddress, password) {
     validationsForCheckUser(emailAddress.trim(), password);
     emailAddress = emailAddress.trim();
     emailAddress = emailAddress.toLowerCase();
-    
+
     let userFromDB = await userCollection.findOne({ emailAddress });
-  
+
     if (!userFromDB) throw "Either the email address or password is invalid";
-  
-    let { firstName, lastName, emailAddress: email, _id, userName } = userFromDB;
-  
+
+    let {
+      firstName,
+      lastName,
+      emailAddress: email,
+      _id,
+      userName,
+    } = userFromDB;
+
     let isValid = await bcrypt.compare(password, userFromDB.password);
-  
-    if (isValid) return { firstName, lastName, emailAddress: email, _id: _id.toString(), userName };
+
+    if (isValid)
+      return {
+        firstName,
+        lastName,
+        emailAddress: email,
+        _id: _id.toString(),
+        userName,
```

Comment:
Why are these changed ???

Reply:
Julien has the prettier formatter on so all these things got changed